godot-core/src/builtin/quaternion.rs

```diff
@@ -54,9 +64,20 @@ impl Quaternion {
         Self::new(x, y, z, w)
     }
 
-    // TODO: Constructors.
-    // pub fn from_vector_vector(arc_to: Vector3, arc_from: Vector3) -> Self {}
-    // pub fn from_basis(basis: Basis) -> Self {}
+    /// Constructs a Quaternion representing the shortest arc between `arc_from` and `arc_to`.
+    ///
+    /// These can be imagined as two points intersecting a unit sphere's surface, with a radius of 1.0.
+    ///
+    // This is an undocumented assumption of Godot's as well.
+    /// The inputs must be unit vectors.
+    ///
+    /// For near-singular cases (`arc_from`≈`arc_to` or `arc_from`≈-`arc_to`) the current implementation is only accurate to about
+    /// 0.001, or better if `double-precision` is enabled.
+    ///
+    /// *Godot equivalent: `Quaternion(arc_from: Vector3, arc_to: Vector3)`*
+    pub fn from_rotation_arc(arc_to: Vector3, arc_from: Vector3) -> Self {
+        <Self as GlamConv>::Glam::from_rotation_arc(arc_to.to_glam(), arc_from.to_glam()).to_front()
+    }
```

Comment:
Oh sorry I missed that order.  Emulating Godot was my intention, just missed that bit.

Debug asserts could be good; I didn't add anything like that since Godot doesn't.  Glam also has a feature for this kind of debug assert, so it might be possible to enable `glam_assert` for debug builds and gain these kinds of assertions for free (though then would the error messages be any good?)

Reply:
I'd probably do our own asserts, then we're more in control -- and conditions for Godot are sometimes also different.

Reply:
Ok, done.